### PR TITLE
More `core:sys/windows` bindings (primarily MiniDump and SHGetKnownFolderPath)

### PR DESCRIPTION
**Pull request by jakubtomsu**
This PR implements the following:
- most of `minidumpapiset.h` (everything you need for reading/writing MiniDumps)
- Known folder IDs
- SHGetKnownFolderPath etc. from `shlObj_core.h` (`shell32.dll`)

All of these things are important for games, in particular `MinuDumpWriteDump` and `SHGetKnownFolderPath(FOLDERID_SavedGames)`

### CODE REVIEW

**Comment by jakubtomsu** [comment]
Hmm, the CI is failing on Linux and MacOS, is that expected since I changed `core:sys/windows`? Or did I accidentally broke something?

**Comment by laytan** [comment]
It also fails on master so you are good

**Comment by jakubtomsu** [comment]
Oh yeah right, thanks for letting me know!

**Comment by karl-zylinski** [comment]
This is great! Having `SHGetKnownFolderPath` means one can get the correct folder to output logs and save games 🥳

File: core/sys/windows/dbghelp.odin [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        Module:            MINIDUMP_MODULE_CALLBACK,
+        IncludeThread:     MINIDUMP_INCLUDE_THREAD_CALLBACK,
+        IncludeModule:     MINIDUMP_INCLUDE_MODULE_CALLBACK,
+        Io:                MINIDUMP_IO_CALLBACK,
+        ReadMemoryFailure: MINIDUMP_READ_MEMORY_FAILURE_CALLBACK,
+        SecondaryFlags:    ULONG,
+        VmQuery:           MINIDUMP_VM_QUERY_CALLBACK,
+        VmPreRead:         MINIDUMP_VM_PRE_READ_CALLBACK,
+        VmPostRead:        MINIDUMP_VM_POST_READ_CALLBACK,
+    },
```
Comment (gingerBill) [inline core/sys/windows/dbghelp.odin:75]: Please use tabs for indentation.

File: core/sys/windows/known_folders.odin [hunk 2]
```diff
@@ -0,0 +1,145 @@
+// +build windows
+package sys_windows
+
+FOLDERID_NetworkFolder:          GUID : {0xD20BEEC4, 0x5CA8, 0x4905, {0xAE, 0x3B, 0xBF, 0x25, 0x1E, 0xA0, 0x9B, 0x53}}
```
Comment (gingerBill) [inline core/sys/windows/known_folders.odin:4]: Prefer the following style:

```odin
FOLDERID_NetworkFolder      :: GUID {0xD20BEEC4, 0x5CA8, 0x4905, {0xAE, 0x3B, 0xBF, 0x25, 0x1E, 0xA0, 0x9B, 0x53}}
```

**Comment by jakubtomsu** [comment]
@gingerBill Thanks for the reviews, it should be fixed now. I keep forgetting about the indentation, sorry😅

File: core/sys/windows/dbghelp.odin [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+MINIDUMP_USER_STREAM :: struct {
+	Type:       ULONG32,
+	BufferSize: ULONG,
+	Buffer:     PVOID,
+}
+
+MINIDUMP_USER_STREAM_INFORMATION :: struct {
+  UserStreamCount: ULONG,
+  UserStreamArray: ^MINIDUMP_USER_STREAM,
```
[the remark this answers is not in the document]
Comment (jakubtomsu) [reply core/sys/windows/dbghelp.odin]: Fixed

**Comment by jakubtomsu** [comment]
@gingerBill This should be resolved and ready to merge, I think all of the indentation issues are gone.

**Pull request merged**